packages/compass-editor/src/json-editor.tsx

```diff
@@ -1267,6 +1267,24 @@ const MultilineEditor = React.forwardRef<EditorRef, MultilineEditorProps>(
       ];
     }, [copyable, formattable, customActions]);
 
+    const getMinLines = useCallback(
+      (minLinesFromProps: number | undefined) => {
+        const defaultMinLines = 10;
+        const requiredMinLinesWithActions = 2;
+        if (minLinesFromProps === undefined) {
+          return defaultMinLines;
+        }
+
+        // We need to have enough height for our action buttons to be accessible
+        if (actions.length) {
+          return Math.max(minLinesFromProps, requiredMinLinesWithActions);
+        }
+
+        return minLinesFromProps;
+      },
+      [actions]
+    );
```

Comment:
I think that's a bit of an overkill, what I was thinking when we were discussing this is literally something like `style={{ minHeight: actions.length > 0 ? <button height + margin> : 'auto' }}` on a container in this component. I don't think we should use minLines for that due to the amount of logic this variable controls and you also don't really get nice alignment with this as 2 lines give you more space than just a button with margins around it takes.

Reply:
Thank you. I wouldn't be able to use button margins as they (buttons) are absolutely positioned but I have addressed this a little differently just now. The min-height of the container now is height of editor container / buttons (24px) + button top position (4px) used as padding for top and bottom on the container to align it in the middle with buttons. Didn't use flex as there was a flicker with the presence of horizontal scrollbar.

Reply:
Thank you for reminding me of this. I just updated my implementation to take care of this without using `minLines`.